File: BroadcastActivity/app/src/main/java/com/example/broadcastactivity/MainActivity.java
package com.example.broadcastactivity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.IntentFilter;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    MyBRReceiver myReceiver;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //核心部分代码：
        myReceiver = new MyBRReceiver();
        IntentFilter itFilter = new IntentFilter();
        itFilter.addAction("android.net.conn.CONNECTIVITY_CHANGE");
        itFilter.addAction("Context.CONNECTIVITY_SERVICE");
        itFilter.addAction("ConnectivityManager.TYPE_MOBILE");
        itFilter.addAction("ConnectivityManager.TYPE_WIFI");
        registerReceiver(myReceiver, itFilter);
    }

    //将广播挂起哦~
    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(myReceiver);
    }
}

File: networktest1/src/main/java/com/example/networktest1/MainActivity.kt
package com.example.networktest1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val webView:WebView=findViewById(R.id.webView)
        val getWebViewBtn:Button=findViewById(R.id.getWebViewBtn)
        webView.settings.javaScriptEnabled=true
        getWebViewBtn.setOnClickListener{
            webView.webViewClient= WebViewClient()
            webView.loadUrl("https://www.baidu.com")
        }

    }
}